Segment-ComScore/Classes/SEGComScoreIntegration.m

```diff
@@ -11,49 +11,54 @@
 
 @implementation SEGComScoreIntegration
 
-- (instancetype)initWithSettings:(NSDictionary *)settings
+- (instancetype)initWithSettings:(NSDictionary *)settings 
 {
     if (self = [super init]) {
         self.settings = settings;
-        self.comScoreClass = [CSComScore class];
+        self.comScore = [SCORAnalytics class];
         
-        [self.comScoreClass setAppContext];
-        SEGLog(@"[CSComScore setAppContext]");
-        [self.comScoreClass setCustomerC2:[self customerC2]];
-        SEGLog(@"[CSComScore setCustomerC2: %@]", [self customerC2]);
-        [self.comScoreClass setPublisherSecret:[self publisherSecret]];
-        SEGLog(@"[CSComScore setPublisherSecret: %@]", [self publisherSecret]);
-        [self.comScoreClass setAppName: [self appName]];
-        SEGLog(@"[CSComScore setAppName: %@]", [self appName]);
-        [self.comScoreClass setSecure: [self useHTTPS]];
-        SEGLog(@"[CSComScore setSecure: %@]", [self useHTTPS] ? @YES : @NO);
-        if ([self autoUpdate]) {
-            if ([self foregroundOnly]) {
-                [self.comScoreClass enableAutoUpdate:[self autoUpdateInterval] foregroundOnly: YES];
-                SEGLog(@"[CSComScore enableAutoUpdate: %@ foregroundOnly: YES]", [self autoUpdateInterval]);
+        SCORPublisherConfiguration *config = [SCORPublisherConfiguration publisherConfigurationWithBuilderBlock:^(SCORPublisherConfigurationBuilder *builder) {
+            // publisherId is also known as c2 value
+            builder.publisherId = settings[@"publisherId"];
+            builder.publisherSecret = settings[@"publisherSecret"];
+            builder.applicationName = settings[@"appName:"];
+            
+            if ([settings[@"autoUpdate"] boolValue] && [settings[@"foregroundOnly"] boolValue]) {
+                builder.usagePropertiesAutoUpdateMode = SCORUsagePropertiesAutoUpdateModeForegroundOnly;
+            } else if ([settings[@"autoUpdate"] boolValue]) {
+                builder.usagePropertiesAutoUpdateMode = SCORUsagePropertiesAutoUpdateModeForegroundAndBackground;
             } else {
-                [self.comScoreClass enableAutoUpdate:[self autoUpdateInterval] foregroundOnly: NO];
-                SEGLog(@"[CSComScore enableAutoUpdate: %@ foregroundOnly: NO]", [self autoUpdateInterval]);
+                builder.usagePropertiesAutoUpdateMode = SCORUsagePropertiesAutoUpdateModeDisabled;
             }
-        }
+            builder.usagePropertiesAutoUpdateInterval = [settings[@"autoUpdateInterval"] integerValue];
+            
+            builder.secureTransmission = [settings[@"useHTTPS"] boolValue];
+            
+            //        TODO: What are these? And do they have equivalents
```

Comment:
please do. I think this is something where it makes a lot of sense to ask them to clarify how to migrate.